Implicit int diagnostics

Starting with C23, implicit-int type specifiers are no longer allowed:
```C
// test.c
static x = 5;
```

```
➜  ~/local/llvm15-release/bin/clang -std=c2x test.c
test.c:1:8: error: a type specifier is required for all declarations
static x = 5;
~~~~~~ ^
1 error generated.
```

This will need a new error diagnostic in `Diagnostics.zig` and an update to `Type.zig` where we issue `.missing_type_specifier`

--- a/src/Diagnostics.zig
+++ b/src/Diagnostics.zig
@@ -374,6 +374,10 @@ const messages = struct {
         const kind = .warning;
         const all = true;
     };
+    pub const missing_type_specifier_c2x = struct {
+        const msg = "a type specifier is required for all declarations";
+        const kind = .@"error";
+    };
     pub const multiple_storage_class = struct {
         const msg = "cannot combine with previous '{s}' declaration specifier";
         const extra = .str;

--- a/src/Type.zig
+++ b/src/Type.zig
@@ -1727,7 +1727,11 @@ pub const Builder = struct {
                     ty = typeof;
                 } else {
                     ty.specifier = .int;
-                    try p.err(.missing_type_specifier);
+                    if (p.comp.langopts.standard.atLeast(.c2x)) {
+                        try p.err(.missing_type_specifier_c2x);
+                    } else {
+                        try p.err(.missing_type_specifier);
+                    }
                 }
             },
             .void => ty.specifier = .void,

--- a/test/cases/missing type specifier.c
+++ b/test/cases/missing type specifier.c
@@ -1,0 +1,4 @@
+//aro-args
+#define EXPECTED_ERRORS \
+    "missing type specifier.c:4:8: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]"
+static x = 5;

--- a/test/cases/missing type specifier_c2x.c
+++ b/test/cases/missing type specifier_c2x.c
@@ -1,0 +1,4 @@
+//aro-args -std=c2x
+#define EXPECTED_ERRORS \
+    "missing type specifier_c2x.c:4:8: error: a type specifier is required for all declarations"
+static x = 5;

